src/state.ts

```diff
@@ -28,8 +28,9 @@ export class StateManager {
   get ignoreCMakeListsMissing(): boolean {
     return this._get<boolean>('ignoreCMakeListsMissing') || false;
   }
-  set ignoreCMakeListsMissing(v: boolean) {
-    this._update('ignoreCMakeListsMissing', v);
+
```

Comment:
Which test this change will fix? I am asking as I am working on some other failing tests.

Reply:
Not anymore since Bob's changes worked around (and thus hided) the real issue.
Also this had been causing test failures in our extension (azure sphere) for a rather long time.

Reply:
A typical issue caused by this is in multi root mode, there's a chance the extension asks for variant before doing configure, even the default is clearly set to 'debug'.

Reply:
So the failure this will fix is the race problem in multi-root? Would you mind explaining that more? Or if that is easier, you can create an issue for the PR and add some details. whichever is more convenient. meanwhile, I am going through all the waits that are being added and trying to figure out if it affects other scenarios.

Reply:
It's hard to explain... The behavior is when the variant is "updated" in global state, it's not actually "completely updated". So in the immediate next "get", the value returned is correct, but the "get" function returns undefined later at certain point, although it's never set to undefined. If you would like to know more, ping me on teams and I'll send you some logs.

Reply:
thank you for the explanation. it helps me understand what the code is doing.

Reply:
I approved the code. I added @andreeis to do the final approval.

Reply:
Thanks for the review!